filament/src/materials/blur.mat

```diff
@@ -0,0 +1,108 @@
+material {
+    name : blur,
+    parameters : [
+        {
+            type : sampler2d,
+            name : ssao,
+            precision: medium
+        },
+        {
+            type : sampler2d,
+            name : depth,
+            precision: medium
+        },
+        {
+            type : float4,
+            name : resolution
+        },
+        {
+            type : int2,
+            name : axis
+        }
+    ],
+    variables : [
+        vertex
+    ],
+    vertexDomain : device,
+    depthWrite : false,
+    depthCulling : true,
+    shadingModel : unlit,
+    variantFilter : [ skinning, shadowReceiver ],
+    culling: none
+}
+
+vertex {
+    void materialVertex(inout MaterialVertexInputs material) {
+        // far-plane in view space
+        vec4 position = getPosition(); // clip-space
+        position.z = 1.0; // far plane
+        material.vertex.xy = (position.xy * 0.5 + 0.5);
+        material.vertex.zw = position.xy;
+    }
+}
+
+fragment {
+    // z-distance (in m) that constitute an edge for bilateral filtering
+    #define EDGE_DISTANCE   0.1
+
+    const int kGaussianCount = 5;
+    const int kRadius = kGaussianCount - 1;
+    const float kGaussianSamples[] = float[](
+        0.239365, 0.199935, 0.116512, 0.0473701, 0.0134367
+    );
+    const float kGaussianWeightSum = 0.993872;
+
+    ivec2 clampToEdge(const sampler2D s, ivec2 uv) {
+        ivec2 size = textureSize(s, 0);
+        return clamp(uv, ivec2(0), size - ivec2(1));
+    }
+
+    float linearizeDepthDifference(float d0, float d1) {
+        mat4 projection = getClipFromViewMatrix();
+        float A = -projection[3].z;
+        float B =  projection[2].z;
+        float K = (2.0 * A) / (d0 * 2.0 + (B - 1.0)); // this is actually a constant
```

Comment:
Is it a constant? `d0` isn't

Reply:
right -- it's a constant for this pixel, i.e. it doesn't depend on the current sample -- it could be calculated outside of the main loop, but it would have to be passed down in each function call. The hope here is that the optimizer will do the right thing based on register pressure, etc...